Git-commit: 924f8f5af31423529cc3940cb2ae9fee736b7517
From: Peter Zijlstra <a.p.zijlstra@chello.nl>
Subject: llist: Add llist_next()
Patch-mainline: 3.2
References: fate#316876

So we don't have to expose the struct list_node member.

Cc: Huang Ying <ying.huang@intel.com>
Cc: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
Link: http://lkml.kernel.org/r/1315836348.26517.41.camel@twins
Signed-off-by: Ingo Molnar <mingo@elte.hu>
[dropped inapplicable change to kernel/irq_work.c]
Acked-by: jbeulich@suse.com

--- a/include/linux/llist.h
+++ b/include/linux/llist.h
@@ -161,6 +161,11 @@ static inline bool llist_empty(const str
 	return ACCESS_ONCE(head->first) == NULL;
 }
 
+static inline struct llist_node *llist_next(struct llist_node *node)
+{
+	return node->next;
+}
+
 extern bool llist_add_batch(struct llist_node *new_first,
 			    struct llist_node *new_last,
 			    struct llist_head *head);
